### DISCUSSION TITLE
Model Validating Same Field Twice?

### DISCUSSION THREAD

**Discussion by jordantshaw** [pydantic/pydantic#3065] [Question] [answered]
Hi All,

I am running into an issue where it appears my model is validating a field twice for some reasons.  The field is a string json value, and I am trying to set the data type to Json.  

I get the following error:
JSON object must be str, bytes or bytearray (type=type_error.json)

So I created a custom validator and tried loading the json using json.loads() but I get the same error.  And I realized that it is validating the field twice for some reason.  So on the first validation, the string is converted to a dict, and then on the second validation, it attempts to load the field (Which is now a dict) as json, resulting in the error.  Ideally I shouldn't even have to use a custom validator for this field, I should be able to just use the Pydantic.Json type.  Any ideas on why this isn't working?

```
class DataSources(BaseModel):
    data_source_id: UUID = None
    key_value: Any = None
    name: str = None
    source_table: str = None

    @validator('key_value', pre=True)
    def key_value_validator(cls, v):
        import json
        return json.loads(v)

    class Config:
        orm_mode = True
        allow_population_by_field_name = True
        alias_generator = utils.snake_to_camel_case
```

**Comment by PrettyWood** [comment]
Hi @jordantshaw
Sorry but I cannot reproduce your error where the validator is called twice.
Please share a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).
Help us so we can help you :)

**Reply by jordantshaw** [reply]
I was actually able to discover the issue.  The reason it was validating twice was a FastAPI issue, not Pydantic. Posted my resolution for others below.

**Comment by jordantshaw** [comment, marked as answer]
I was able to figure out what the issue was.  The issue was not with Pydantic, but it was the way I was using the models in FastAPI.  Fl anyone else who may have this issue, the problem was I was returning the Pydantic model in the Fast API endpoint, however, I was also setting the model validation in the "response_model" parameter in the endpoint itself. So it was validating twice, once in my endpoint when returning the data as a model, and then once again before the data was sent back to the user due to "response_model".  To fix this I simply started returning my data as a dictionary instead.
